The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/runs/6051955003?check_suite_focus=true) [(plain)](https://github.com/rust-lang/rust/commit/7d9a278a7e381279779cb818283788a785d26a28/checks/6051955003/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
    Checking rand v0.7.3
    Checking std v0.0.0 (/checkout/library/std)
    Checking alloc v0.0.0 (/checkout/library/alloc)
    Checking core v0.0.0 (/checkout/library/core)
error[E0658]: use of unstable library feature 'split_as_slice'
     |
     |
2346 |     assert_eq!(split.as_slice(), &[1, 2, 3, 4, 5, 6]);
     |
     = note: see issue #96137 <https://github.com/rust-lang/rust/issues/96137> for more information
     = help: add `#![feature(split_as_slice)]` to the crate attributes to enable


error[E0658]: use of unstable library feature 'split_as_slice'
     |
     |
2348 |     assert_eq!(split.as_slice(), &[3, 4, 5, 6]);
     |
     = note: see issue #96137 <https://github.com/rust-lang/rust/issues/96137> for more information
     = help: add `#![feature(split_as_slice)]` to the crate attributes to enable


error[E0658]: use of unstable library feature 'split_as_slice'
     |
     |
2351 |     assert_eq!(split.as_slice(), &[]);
     |
     = note: see issue #96137 <https://github.com/rust-lang/rust/issues/96137> for more information
     = help: add `#![feature(split_as_slice)]` to the crate attributes to enable

```

</details>

--- a/library/core/src/slice/iter.rs
+++ b/library/core/src/slice/iter.rs
@@ -358,6 +358,20 @@ impl<'a, T: 'a, P: FnMut(&T) -> bool> Split<'a, T, P> {
     pub(super) fn new(slice: &'a [T], pred: P) -> Self {
         Self { v: slice, pred, finished: false }
     }
+    /// Returns a slice which contains items not yet handled by split.
+    /// # Example
+    ///
+    /// ```
+    /// #![feature(split_as_slice)]
+    /// let slice = [1,2,3,4,5];
+    /// let mut split = slice.split(|v| v % 2 == 0);
+    /// assert!(split.next().is_some());
+    /// assert_eq!(split.as_slice(), &[3,4,5]);
+    /// ```
+    #[unstable(feature = "split_as_slice", issue = "96137")]
+    pub fn as_slice(&self) -> &'a [T] {
+        if self.finished { &[] } else { &self.v }
+    }
 }
 
 #[stable(feature = "core_impl_debug", since = "1.9.0")]

--- a/library/core/tests/lib.rs
+++ b/library/core/tests/lib.rs
@@ -46,6 +46,7 @@
 #![feature(sort_internals)]
 #![feature(slice_take)]
 #![feature(slice_from_ptr_range)]
+#![feature(split_as_slice)]
 #![feature(maybe_uninit_uninit_array)]
 #![feature(maybe_uninit_array_assume_init)]
 #![feature(maybe_uninit_write_slice)]

--- a/library/core/tests/slice.rs
+++ b/library/core/tests/slice.rs
@@ -2339,6 +2339,18 @@ fn slice_rsplit_array_mut() {
     }
 }
 
+#[test]
+fn split_as_slice() {
+    let arr = [1, 2, 3, 4, 5, 6];
+    let mut split = arr.split(|v| v % 2 == 0);
+    assert_eq!(split.as_slice(), &[1, 2, 3, 4, 5, 6]);
+    assert!(split.next().is_some());
+    assert_eq!(split.as_slice(), &[3, 4, 5, 6]);
+    assert!(split.next().is_some());
+    assert!(split.next().is_some());
+    assert_eq!(split.as_slice(), &[]);
+}
+
 #[should_panic]
 #[test]
 fn slice_split_array_ref_out_of_bounds() {

